hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -924,6 +964,13 @@ extends Importer:
           case S(sym: BlockMemberSymbol) => sym.modTree.exists(isModule)
           case _ => false
   
+    def isModuleMethodApp(t: Term): Bool = t match
```

Comment:
Better follow the "parse, don't validate" methodology (https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/).

Instead of returning a Boolean, just return an option of the relevant information so that info doesn't have to be retrieved separately in a way that might not be perfectly in sync, creating an unnecessary source of bugs.